### ISSUE TITLE
Use empty data instead of null in `GET /api/v1/rules` response data

### ISSUE THREAD

**Issue by kakkotetsu** [VictoriaMetrics/VictoriaMetrics#4221]
### Is your feature request related to a problem? Please describe

VictoriaMetrics(vmalert) uses `null` value on `GET /api/v1/rules` response data.
as follows.

`groups` is `null`.
```json
{
  "status": "success",
  "data": {
    "groups": null
  }
}
```

`rules` is `null`.
```json
      {
        "name": "mds",
        "rules": null,
        "interval": 60,
        "lastEvaluation": "2023-04-28T03:12:17.320533579Z",
        "type": "prometheus",
        "id": "5324614860393267221",
        "file": "/etc/alerts/alert.rules.ceph.yml",
        "concurrency": 1
      }
```

On the other hand, Prometheus uses an empty value.

`rules` is `[]`.
```json
      {
        "name": "mds",
        "file": "/etc/prometheus/alert.rules.ceph.yml",
        "rules": [],
        "interval": 15,
        "limit": 0,
        "evaluationTime": 3.665e-06,
        "lastEvaluation": "2023-04-28T03:12:11.913780266Z"
      }
```

Both formats are considered correct as JSON format.
But on Grafana `9.5.1` "Alert tules" can't handle `null` value...  ( https://github.com/grafana/grafana/issues/67463 )

It is correct that Grafana should be able to handle null values correctly, I also think it would be good for VictoriaMetrics to use the same format as Prometheus.

### Describe the solution you'd like

Use empty data instead of null in `GET /api/v1/rules` response data.

**Comment by hagen1778** [comment]
Thanks for reporting this, @kakkotetsu!
The response format was fixed in https://github.com/VictoriaMetrics/VictoriaMetrics/commit/3383f12a4b44cef62596414207b60b74839206e7 and will be included in the next release.

Meanwhile, you can try [building vmalert from sources](https://docs.victoriametrics.com/vmalert.html#how-to-build-from-sources).

**Comment by valyala** [comment]
[vmalert](https://docs.victoriametrics.com/vmalert.html) should properly return empty list instead of `null` in the response from `/api/v1/rules` starting from [v1.91.0](https://docs.victoriametrics.com/CHANGELOG.html).

Closing the issue as fixed.

**Comment by valyala** [comment]
The bugfix has been backported to [v1.87.6 LTS release](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.6).

**Comment by hagen1778** [comment]
The fix was included in https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.91.1

**Comment by valyala** [comment]
FYI, the bugfix has been also included in [the long-term support release v1.79.14](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.79.14).

**Comment by valyala** [comment]
FYI, the bugfix has been also included in [the long-term support release v1.87.7](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.87.7).
